storage/engine/rocksdb/db.cc

```diff
@@ -159,6 +160,38 @@ rocksdb::Slice ToSlice(DBString s) {
   return rocksdb::Slice(s.data, s.len);
 }
 
+std::string EncodeKey(DBKey k) {
+  std::string s;
+  EncodeBytes(&s, k.key.data, k.key.len);
+  if (k.walltime != 0 || k.logical != 0) {
+    EncodeUint64Decreasing(&s, uint64_t(k.walltime));
+    EncodeUint32Decreasing(&s, uint32_t(k.logical));
+  }
+  return s;
+}
+
+const int mvcc_version_timestamp_size = 12;
+
+bool DecodeKey(rocksdb::Slice buf, std::string *key, int64_t *walltime, int32_t *logical) {
+  key->clear();
+  if (!DecodeBytes(&buf, key)) {
+    return false;
+  }
+  if (buf.size() == mvcc_version_timestamp_size) {
+    uint64_t w;
+    if (!DecodeUint64Decreasing(&buf, &w)) {
+      return false;
+    }
+    uint32_t l;
+    if (!DecodeUint32Decreasing(&buf, &l)) {
+      return false;
+    }
+    *walltime = int64_t(w);
+    *logical = int32_t(l);
+  }
+  return true;
```

Comment:
I think this should return false if `buf.size()` is not either `mvcc_version_timestamp_size` or zero.